tests/test_plot_manager.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from .common import test_result_collector as trc
+from .common.test_utils import convert_to_bytes, ROOT_DIR
+from .mocks.mock_config import MockConfig
+
+from model_analyzer.cli.cli import CLI
+from model_analyzer.config.input.config_command_analyze \
+    import ConfigCommandAnalyze
+
+from model_analyzer.plots.plot_manager import PlotManager
+from model_analyzer.result.result_manager import ResultManager
+from model_analyzer.state.analyzer_state_manager import AnalyzerStateManager
+
+from filecmp import cmp
+from shutil import rmtree
+from unittest.mock import MagicMock, patch
+
+import pickle
```

Comment:
https://medium.com/ochrona/python-pickle-is-notoriously-insecure-d6651f1974c9#:~:text=Pickle%20is%20dangerous%20because%20during,__%20method%20to%20be%20executed.  

bah! we are likely getting hit with this vulnerability even if its just test code.
Is there a different serialization/deserialization method we can use?

Reply:
I think we're actually safe here. Taken from the article you linked:
`Utilize pickle, but only with strict controls in place such as only losing pickles that you created yourself, or only loading pickles from trusted parties where cryptographic hashes can be used to verify that the source is authentic and that the pickle hasn’t been modified.`

In this case I've hardcoded the file we are loading into the unit test (unlike say a checkpoint file that a user would provide) and saved it in the repo.

Reply:
It looks like _compare_plot_data is only comparing simple primitives, so I don't think it would be too difficult to compare to a JSON serialization or similar to avoid pickle usage altogether if concerned.

Reply:
The issue is that the SimplePlot class is not JSON serializable. So, this doesn't work:
`json.dumps(plot_manager._simple_plots['add_sub']
                         ['throughput_v_latency'].__dict__)`
                        
The issue is the matlab.pyplot being stored in the SimplePlot.

Reply:
Right, but if you're hard-coding a golden file, you don't need to dump the object, you can manually construct/print a dict once.

Or if you implement the __dict__, you can exclude the non-serializable objects like the plots if not required (plots aren't being compared in your comparison)

Reply:
I've replaced pickle with json